### PR TITLE
PTCD-738 Delete T Level

### PR DESCRIPTION
**Pull request by gunndabad**

### CODE REVIEW

File: src/Dfc.CourseDirectory.WebV2/Features/TLevels/DeleteTLevel/View.cshtml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+@section Scripts {
+    <script type="text/javascript">
+        function enableDeleteBtn() {
+            $('#pttcd-delete-tlevel__delete-btn').attr(
+                'disabled',
+                $('#Confirm').is(':checked') ? null : 'disabled');
+        }
+
+        // timeout is required otherwise the _checked_ property check doesn't return the correct value
+        // (maybe GDS scripts need chance to run?)
```
Comment (timabell) [inline src/Dfc.CourseDirectory.WebV2/Features/TLevels/DeleteTLevel/View.cshtml:80]: I'm not sure I'd leave the question in here, don't think it adds much

File: src/Dfc.CourseDirectory.WebV2/ProviderOwnershipCache.cs [hunk 2]
```diff
@@ -98,20 +128,29 @@ public void OnCourseDeleted(Guid courseId)
...
         private static string GetCourseCacheKey(Guid courseId) =>
-            $"course-ukprns:{courseId}";
+            $"course-providers:{courseId}";
+
+        private static string GetTLevelCacheKey(Guid tLevelId) =>
+            $"tlevel-providers:{tLevelId}";
 
         private static string GetVenueCacheKey(Guid venueId) =>
-            $"venue-ukprns:{venueId}";
+            $"venue-providers:{venueId}";
```
Comment (timabell) [inline src/Dfc.CourseDirectory.WebV2/ProviderOwnershipCache.cs:153]: What's the impact of changing cache keys on prod? Presumably the old stuff will expire out and the new stuff has no clashes?

File: src/Dfc.CourseDirectory.WebV2/ServiceCollectionExtensions.cs [hunk 3]
```diff
@@ -118,7 +118,7 @@ public static class ServiceCollectionExtensions
             services.AddFormFlow();
             services.AddJourneyStateTypes(typeof(ServiceCollectionExtensions).Assembly);
 
-            services.AddSingleton<IProviderOwnershipCache, ProviderOwnershipCache>();
+            services.AddTransient<IProviderOwnershipCache, ProviderOwnershipCache>();
```
Comment (timabell) [inline src/Dfc.CourseDirectory.WebV2/ServiceCollectionExtensions.cs:121]: Why the change to transient (I think this is right but just checking)? Where does the cache actually live? I see it's injected.

File: tests/Dfc.CourseDirectory.WebV2.Tests/FeatureTests/TLevels/DeleteTLevelTests.cs [hunk 4]
```diff
@@ -0,0 +1,487 @@
...
+            JourneyModel journeyModel = null)
+        {
+            return CreateJourneyInstance(
+                "DeleteTLevel",
+                configureKeys: keysBuilder => keysBuilder
+                    .With("tLevelId", tLevelId),
+                journeyModel ?? new JourneyModel());
+        }
+    }
+}
```
Comment (timabell) [inline tests/Dfc.CourseDirectory.WebV2.Tests/FeatureTests/TLevels/DeleteTLevelTests.cs:487]: This is quite a big test file, is there any opportunity for good old fashioned DRY to make the intent clearer with less lines of code?

**Review by timabell — APPROVED** [review]

File: src/Dfc.CourseDirectory.WebV2/ProviderOwnershipCache.cs (hunk 2 above)
Comment (gunndabad) [reply src/Dfc.CourseDirectory.WebV2/ProviderOwnershipCache.cs:153]: Yep

File: src/Dfc.CourseDirectory.WebV2/ServiceCollectionExtensions.cs (hunk 3 above)
Comment (gunndabad) [reply src/Dfc.CourseDirectory.WebV2/ServiceCollectionExtensions.cs:121]: It now has a dependency on `ISqlQueryDispatcher` and that guy isn't `Singleton`
